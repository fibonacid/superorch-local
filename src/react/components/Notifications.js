import React from 'react';
import PropTypes from 'prop-types';
import FlashMessage from "./FlashMessage";

function Notifications(props) {
  return (
    <ul data-test={'NotificationsComponent'}>
      {props.notifications.map((notification, i) => (
        <FlashMessage key={i}>
          {notification.message}
        </FlashMessage>
      ))}
    </ul>
  )
}

Notifications.propTypes = {
  notifications: PropTypes.arrayOf(PropTypes.object)
};

export default Notifications;
