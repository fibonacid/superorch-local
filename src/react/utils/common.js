import _ from "lodash";

/**
 * This function is used to transform
 * all the prop keys to lower case.
 * By doing so you can avoid errors like this:
 *
 * React does not recognize the `contentState` prop on a DOM element.
 * If you intentionally want it to appear in the DOM as a custom attribute,
 * spell it as lowercase `contentstate` instead.
 * If you accidentally passed it from a parent component, remove it from the DOM element.
 *
 * @param props
 * @returns {{}}
 */
export function lowerCasedProps(obj) {
  return _.mapKeys(obj, (value, key) => {
    return key.toLowerCase()
  });
}
