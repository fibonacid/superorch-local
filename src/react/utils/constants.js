export const INITIAL_USER_ID = -1;
