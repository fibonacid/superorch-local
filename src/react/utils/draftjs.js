import { OrderedMap, List, Map } from "immutable";

/**
 * Function returns collection of currently selected blocks.
 */
export function getSelectedBlocksMap(editorState: EditorState): OrderedMap {
  const selectionState = editorState.getSelection();
  const contentState = editorState.getCurrentContent();
  const startKey = selectionState.getStartKey();
  const endKey = selectionState.getEndKey();
  const blockMap = contentState.getBlockMap();
  return blockMap
    .toSeq()
    .skipUntil((_, k) => k === startKey)
    .takeUntil((_, k) => k === endKey)
    .concat([[endKey, blockMap.get(endKey)]]);
}

/**
 * Function returns collection of currently selected blocks.
 */
export function getSelectedBlocksList(editorState: EditorState): List {
  return getSelectedBlocksMap(editorState).toList();
}

export function getSelectedTextBlocks(editorState: EditorState): string {
  const selectedBlocks = getSelectedBlocksList(editorState);
  let text;
  selectedBlocks.forEach((block, i) => {
    text = `${block.getText()}${
      i < selectedBlocks.size
        ? '\n'
        : ''
    }`;
  });
  return text;
}

export function findWithRegex(regex, contentBlock, callback) {
  const text = contentBlock.getText();
  let matchArr, start;
  while ((matchArr = regex.exec(text)) !== null) {
    start = matchArr.index;
    callback(start, start + matchArr[0].length);
  }
}
