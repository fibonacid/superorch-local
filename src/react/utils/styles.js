export function resetAppearance() {
  return (`
  &:focus,
  &:hover,
  &:active {
    outline:0px !important;
    -webkit-appearance:none;
    box-shadow: none !important;
  }
  `)
}
