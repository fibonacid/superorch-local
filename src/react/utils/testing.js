import { checkPropTypes } from "prop-types";

export const testFunction = () => {};

export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
  return propsErr;
};
