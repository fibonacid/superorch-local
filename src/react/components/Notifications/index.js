import {connect} from "react-redux";
import Notifications from "./Notifications";

const mapStateToProps = state => ({
  notifications: state.flash
});

export default connect(mapStateToProps)(Notifications)
