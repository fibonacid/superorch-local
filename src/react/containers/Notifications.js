import {connect} from "react-redux";
import Notifications from "../components/Notifications";

const mapStateToProps = state => ({
  notifications: state.flash
});

export default connect(mapStateToProps)(Notifications)
