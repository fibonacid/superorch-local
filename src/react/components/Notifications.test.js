import React from "react";
import {shallow} from 'enzyme'
import Notifications from "./Notifications";
import {findByTestAttr} from "../utils/testing";
import toJson from "enzyme-to-json";

const setUp = (props={}) => {
  return shallow(<Notifications {...props} />);
};

describe('Notifications Component', () => {

  describe('When has props', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = setUp({
       notifications: [
         { message: "First Message", type: "TEST" },
         { message: "Second Message", type: "TEST" }
       ]
      })
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('Should renderer without errors', () => {
      const component = findByTestAttr(wrapper, 'NotificationsComponent');
      expect(component.length).toBe(1);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('When has NO props', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = setUp()
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('Should not render', () => {}).todo();
  });
});
