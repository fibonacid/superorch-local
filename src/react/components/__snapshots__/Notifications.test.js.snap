// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Component When has props Should renderer without errors 1`] = `
<Notificationsstyles__StyledList
  data-test="NotificationsComponent"
>
  <Notificationsstyles__StyledItem
    data-test="list-item"
    key="0"
  >
    First Message
  </Notificationsstyles__StyledItem>
  <Notificationsstyles__StyledItem
    data-test="list-item"
    key="1"
  >
    Second Message
  </Notificationsstyles__StyledItem>
</Notificationsstyles__StyledList>
`;
